libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1137,21 +1120,128 @@ void GCS_MAVLINK::send_ahrs(AP_AHRS &ahrs)
+    // to clear the bit and send quickly but slower links to still store the string. Regardless of mixed
+    // bitrates of ports, a maximum of GCS_MAVLINK_PAYLOAD_STATUS_CAPACITY strings can be buffered. Downside
+    // is if you have a super slow link mixed with a faster port, if there are GCS_MAVLINK_PAYLOAD_STATUS_CAPACITY
+    // string in the slow queue then a 6th item can not be queued for the faster link
+
+    dest_bitmask &= mavlink_active;
+    if (!dest_bitmask) {
+        // nowhere to send
+        return;
+    }
+
+    statustext_t statustext{};
+    statustext.bitmask = mavlink_active & dest_bitmask;
+    statustext.msg.severity = severity;
+    memcpy(statustext.msg.text, text, sizeof(statustext.msg.text));
+    _statustext_queue.push(statustext);
+}
+
+/*
+  send a statustext message to specific MAVLink connections in a bitmask
+ */
+void GCS_MAVLINK::retry_statustext(void)
+{
+    // create bitmask of what mavlink ports we should send this text to.
+    // note, if sending to all ports, we only need to store the bitmask for each and the string only once.
+    // once we send over a link, clear the port but other busy ports bit may stay allowing for faster links
+    // to clear the bit and send quickly but slower links to still store the string. Regardless of mixed
+    // bitrates of ports, a maximum of GCS_MAVLINK_PAYLOAD_STATUS_CAPACITY strings can be buffered. Downside
+    // is if you have a super slow link mixed with a faster port, if there are GCS_MAVLINK_PAYLOAD_STATUS_CAPACITY
+    // string in the slow queue then a 6th item can not be queued for the faster link
+
+    if (_statustext_queue.empty()) {
+        // nothing to do
+        return;
+    }
+
+//    if (_statustext_queue.available >= 2) {
+//        // TODO: reorder _statustext_queue for priority. Higher priority msgs skips ahead in line
+//    }
+
+    // keep sending until all channels have unable to send (full) or have nothing to send
+    bool still_sending = true;
+
+    // using 'while' to ensure we either run out of text to send or all buffers on all ports are full
+    while (still_sending) {
+        still_sending = false; // if we send something, keep going
+
+        // populate statustext
+        statustext_t statustext; // no need to {} init it because the pop will memcpy all bytes and when it was loaded it was properly zero initialized
+        if (!_statustext_queue.pop(statustext)) {
+            // all done, nothing else to send
+            return;
+        }
+
+        // try and send to all active mavlink ports listed in the statustext.bitmask
+        for (uint8_t i=0; i<MAVLINK_COMM_NUM_BUFFERS; i++) {
+
+            uint8_t chan_bit = (1<<i);
+            // logical AND (&) to mask them all together
+            if (statustext.bitmask & // something is queued
+                    mavlink_active & // to an active port
+                    chan_bit) {      // and that's the port index we're looping at
```

Comment:
oh sorry, I forgot they all funnel through it. yeah this is OK to change by removing the mavlink_active check in the loop